lib/jnpr/junos/utils/start_shell.py

```diff
@@ -21,28 +22,39 @@ def _ssh_exec(self, command):
 
     """
 
-    def __init__(self, nc):
+    def __init__(self, nc, timeout=30):
```

Comment:
I disagree with this. nc is just passed to take user credentials. so a user need not to pass nc/dev with a connection opened with device. 
for ex:
```
dev = Device('choc-mx240-k', user='regress', password='MaRtInI', gather_facts=False)
with StartShell(dev) as ss:
    print ss.run('cli show version', '---\(more\s?\d*%?\)---|%')
```
So if you notice dev.open() is not required here.